.travis.yml

```diff
@@ -2,4 +2,30 @@ language: java
 sudo: false
 cache:
   directories:
-      - $HOME/.m2/repository
+    - "$HOME/.m2/repository"
+
+install: mvn install -DskipTests=true -Dmaven.javadoc.skip=true
+script:
+  - jdk_switcher use oraclejdk8
+  - mvn test
+  - jdk_switcher use oraclejdk7
+  - mvn test
+before_deploy: bin/build-release.sh
```

Comment:
Maybe we should move the running of `bin/build-release.sh` to be the last step of the `script` section?

My thinking:  I'm a bit worried about a scenario where there is a bug introduced in `bin/build-release.sh` by a PR, but we don't catch it during the PR's travis builds since it is *implied* by the travis docs that all of the deploy-related steps won't be run until we merge into master.  (The [docs](https://docs.travis-ci.com/user/deployment#Pull-Requests) actually just say the deployment step won't be done for PR builds, and don't specifically say whether the `before_deploy` step would be run for a PR build.)